[macOS] Linker error with new Skia build

- Operating System: macOS 10.15
- Revery Version: #395ed2ed

- Steps to reproduce:
  - Compile revery on macOS 10.15

- Actual Result:
  - The following linking error:
```
Undefined symbols for architecture x86_64:
  "_jpeg_crop_scanline", referenced from:
      SkJpegCodec::onStartScanlineDecode(SkImageInfo const&, SkCodec::Options const&) in libskia.a(jpeg.SkJpegCodec.o)
  "_jpeg_skip_scanlines", referenced from:
      SkJpegCodec::onSkipScanlines(int) in libskia.a(jpeg.SkJpegCodec.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
File "caml_startup", line 1:
Error: Error during linking
error: command failed: 'dune' 'build' '-p' 'Revery' '-j4' (exited with 1)
esy-build-package: exiting with errors above...
error: build failed with exit code: 1
```
- Expected Result:
  - Revery compiles


@bryphe was saying this might be due to something with the `discover.ml` of `reason-harfbuzz`, so that might be a good place to start looking.

--- a/src/config/discover.re
+++ b/src/config/discover.re
@@ -41,8 +41,13 @@ let () =
         }
       };
 
+    let join = String.concat("|");
+    print_endline("conf.cflags: " ++ join(conf.cflags));
+    print_endline("conf.libs: " ++ join(conf.libs));
+
     let ccopt = s => ["-ccopt", s];
     let cclib = s => ["-cclib", s];
+    let framework = s => ["-framework", s];
     let flags =
       switch (get_os) {
       | Linux =>
@@ -66,6 +71,10 @@ let () =
 
     let cflags =
       switch (get_os) {
+      | Mac =>
+        []
+        @ ["-I" ++ Sys.getenv("SKIA_INCLUDE_PATH")]
+        @ ["-I" ++ Sys.getenv("SKIA_INCLUDE_PATH") ++ "/c"]
       | Linux =>
         //conf.cflags
         []
@@ -81,6 +90,18 @@ let () =
 
     let libs =
       switch (get_os) {
+      | Mac =>
+        []
+        @ ["-L" ++ Sys.getenv("JPEG_LIB_PATH")]
+        @ ["-L" ++ Sys.getenv("SKIA_LIB_PATH")]
+        @ ["-L" ++ Sys.getenv("FREETYPE2_LIB_PATH")]
+        @ framework("CoreServices")
+        @ framework("CoreGraphics")
+        @ framework("CoreText")
+        @ framework("CoreFoundation")
+        @ ["-lskia"]
+        @ ["-lstdc++"]
+        @ [Sys.getenv("JPEG_LIB_PATH") ++ "/libturbojpeg.a"]
       | Linux =>
         //conf.libs
         []

--- a/src/config/discover.re
+++ b/src/config/discover.re
@@ -41,10 +41,6 @@ let () =
         }
       };
 
-    let join = String.concat("|");
-    print_endline("conf.cflags: " ++ join(conf.cflags));
-    print_endline("conf.libs: " ++ join(conf.libs));
-
     let ccopt = s => ["-ccopt", s];
     let cclib = s => ["-cclib", s];
     let framework = s => ["-framework", s];

